dali/kernels/signal/decibel/to_decibels_gpu.cu

```diff
@@ -0,0 +1,110 @@
+#include "dali/core/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/core/format.h"
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/signal/decibel/decibel_calculator.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+
+template <typename T>
+struct SampleDesc {
+  const T *in = nullptr;
+  T *out = nullptr;
+  int64_t size = 0;
+};
+
+template <typename T>
+__global__ void ToDecibelsKernel(const SampleDesc<T>* sample_data,
+                                 ToDecibelsArgs<T> args,
+                                 const T* max_values = nullptr) {
+  const int64_t block_size = blockDim.y * blockDim.x;
+  const int64_t grid_size = gridDim.x * block_size;
+  const int sample_idx = blockIdx.y;
+  const auto& sample = sample_data[sample_idx];
+  const int64_t offset = block_size * blockIdx.x;
+  const int64_t tid = threadIdx.y * blockDim.x + threadIdx.x;
+  assert(args.ref_max == (max_values != nullptr));
+  T s_ref = args.ref_max ? max_values[sample_idx] : args.s_ref;
+  DecibelCalculator<T> dB(args.multiplier, s_ref, args.min_ratio);
+  for (int64_t idx = offset + tid; idx < sample.size; idx += grid_size) {
+    sample.out[idx] = dB(sample.in[idx]);
+  }
+}
+
+template <typename T, int Dims>
+ToDecibelsGpu<T, Dims>::~ToDecibelsGpu() = default;
+
+template <typename T, int Dims>
+KernelRequirements ToDecibelsGpu<T, Dims>::Setup(KernelContext &context,
+                                                 const InListGPU<T, Dims> &in,
+                                                 const ToDecibelsArgs<T> &args,
+                                                 span<const T> max_values) {
+  auto out_shape = in.shape;
+  const size_t num_samples = in.size();
+  ScratchpadEstimator se;
+  se.add<SampleDesc<T>>(AllocType::Host, num_samples);
+  se.add<SampleDesc<T>>(AllocType::GPU, num_samples);
+  KernelRequirements req;
+  req.scratch_sizes = se.sizes;
+  req.output_shapes = {out_shape};
+  return req;
+}
+
+template <typename T, int Dims>
+void ToDecibelsGpu<T, Dims>::Run(KernelContext &context, const OutListGPU<T, Dims> &out,
+                                 const InListGPU<T, Dims> &in, const ToDecibelsArgs<T> &args,
+                                 span<const T> max_values) {
+  auto num_samples = in.size();
+  auto* sample_data = context.scratchpad->Allocate<SampleDesc<T>>(AllocType::Host, num_samples);
+
+  for (int i = 0; i < num_samples; i++) {
+    auto &sample = sample_data[i];
+    sample.out = out.tensor_data(i);
+    sample.in = in.tensor_data(i);
+    sample.size = volume(in.tensor_shape(i));
+    assert(sample.size == volume(out.tensor_shape(i)));
+  }
+
+  auto* sample_data_gpu = context.scratchpad->Allocate<SampleDesc<T>>(AllocType::GPU, num_samples);
+  cudaMemcpyAsync(sample_data_gpu, sample_data, num_samples * sizeof(SampleDesc<T>),
+                  cudaMemcpyHostToDevice, context.gpu.stream);
+  cudaDeviceSynchronize();
+  dim3 block(32, 32);
+  dim3 grid(32, num_samples);
```

Comment:
You could risk something like:
```suggestion
  dim3 grid(max(32, 1024/num_samples), num_samples);
```
This would keep SM load balanced for smaller batches.